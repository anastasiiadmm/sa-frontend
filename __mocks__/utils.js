import * as reduxHooks from 'react-redux';

jest.mock('../src/redux/store');
jest.mock('react-redux');

export const mockedUseSelectors = jest.spyOn(reduxHooks, 'useSelector');
export const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
