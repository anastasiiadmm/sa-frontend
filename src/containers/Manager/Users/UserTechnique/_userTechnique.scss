@use 'assets/scss/_mixins.scss';

.UserTechnique {
  position: relative;
  &__table {
    padding: 16px;
    background: #ffffff;
    border-radius: 0 0 12px 12px;
  }

  &__btn_handler {
    position: absolute;
    right: 16px;
    top: 84px;
  }

  &__image-styles {
    width: 48px;
    height: 50px;
    overflow: clip;
    border-radius: 3px;
    object-fit: cover;
  }

  &__mobile_block {
    display: none;
  }

  &__arrow-left {
    padding: 0 10px;
    cursor: pointer;
    width: 35px;
  }

  &__header {
    margin-bottom: 60px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__header-title {
    display: flex;
    align-items: baseline;
  }

  &__title {
    display: flex;
    font-weight: 400 !important;
  }

  &__list_table_mobile {
    display: none;
  }

  &__view-button {
    background: #F8F8F8;
    border: none;
    color: #689F3A;
    font-weight: 700;
    font-size: 14px;
    margin-left: 12px;
  }

  &__subtitle {
    margin: 0;
    font-weight: 700 !important;
  }

  &__tractor-blue {
    width: 24px;
  }

  &__text_processed_area {
    position: absolute;
    left: 20px;
  }

  &__btn_table {
    background-color: #E4F0FD;
    width: 36px;
    height: 36px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 8px;
    margin-right: 10px;
  }

  &__deleteBg {
    background-color: #FFE2E2;
  }

  @include mixins.bp(tablet) {
    &__title {
      display: block;
    }

    &__subtitle {
      margin-bottom: 0 !important;
    }

    &__btn_handler {
      position: initial;
    }

    &__header {
      flex-direction: column;
      align-items: initial;
      padding-left: 7px;
    }
  }

  @include mixins.bp(mobile) {
    &__header-title {
      display: block;
    }
    &__header {
      display: block;
    }

    &__btn_handler {
      position: relative;
      top: 0;
      left: 0;
      > button {
        width: 100%;
      }

      a {
        > button {
          margin-left: 0;
          width: 100%;
          margin-top: 15px;
        }
      }
    }
    &__title {
      display: block;
      margin-top: 12px;
    }
    &__subtitle {
      padding-left: 0;
    }
  }

  .ant-typography h3 {
    margin: 0;
  }
}

@media screen and (max-width: 990px) {
  .UserTechnique {
    &__btn_handler {
      position: absolute;
      left: 16px;
      top: 165px;

      > button {
        width: 192px;
        height: 40px;
        font-weight: 600;
        font-size: 12px;
      }
    }
    &__view-button {
      width: 192px;
      height: 34px;
      font-size: 13px;
    }

    &__subtitle {
      margin-top: -26px;
    }
  }
}

@media screen and (max-width: 620px) {
  .UserTechnique {
    &__header-title {
      display: block;
    }
    &__header {
      display: block;
    }

    &__table_list {
      display: none;
    }

    &__header {
      margin-bottom: 10px;
    }
    &__list_table_mobile {
      display: block;
      margin: 20px;
      > div {
        background-color: #ffffff;
        border-radius: 12px;
        margin-top: 20px;
        padding: 25px 15px 20px 15px;
      }

      .block_avatar {
        display: flex;
        > div {
          > img {
            width: 32px;
            height: 32px;
            border-radius: 8px;
          }
        }

        .block_title {
          margin-top: -13px;
          padding-left: 16px;
          > p {
            line-height: 0.5;
            > b {
              font-weight: 500;
              color: #000000;
            }
          }
          .todo {
            font-size: 12px;
            font-weight: 400;
            color: #79859E;
          }
        }
      }
    }

    &__subtitle {
      margin-top: 0px;
    }

    &__mobile_block {
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      grid-column-gap: 10px;
      grid-row-gap: 10px;

      > button {
        display: flex;
        align-items: center;
        border: none;
        width: 100%;
        height: 104px;
        background-color: #F8F8F8;
        border-radius: 12px;
        padding: 14px;
        cursor: pointer;
        > div {
          > p {
            height: 42px;
            font-weight: 600;
            margin-top: -5px;
            margin-bottom: 0;
            text-align: left;
            font-size: 13px;
          }
        }
      }
    }
    &__btn_handler {
      display: none;
      position: relative;
      top: 0;
      left: 0;
      > button {
        width: 100%;
      }

      a {
        > button {
          margin-left: 0;
          width: 100%;
          margin-top: 15px;
        }
      }
    }
    &__title {
      display: block;
      margin-top: 12px;
    }
    &__subtitle {
      padding-left: 0;
    }
  }

  .ant-typography h3 {
    margin: 0;
  }
}

.link-icons {
  cursor: pointer;
}

