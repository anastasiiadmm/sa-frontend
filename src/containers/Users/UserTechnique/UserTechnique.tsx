import React from 'react';

const UserTechnique = () => {
  return <div>UserTechnique</div>;
};

export default UserTechnique;
