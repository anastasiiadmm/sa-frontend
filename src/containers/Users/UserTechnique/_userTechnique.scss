.UserTechnique {
  &__table {
    padding: 16px;
    background: #ffffff;
  }

  &__arrow-left {
    padding: 0 10px;
    cursor: pointer;
    width: 35px;
  }

  &__header {
    margin-bottom: 33px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__header-title {
    display: flex;
    align-items: baseline;
  }

  &__title {
    display: flex;
    font-weight: 400 !important;
  }

  &__subtitle {
    margin: 0;
    padding-left: 7px;
    font-weight: 700 !important;
  }
}
