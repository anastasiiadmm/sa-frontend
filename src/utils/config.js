export const apiURL = '';

export const defaultError = { detail: 'На сервере что-то пошло не так' };
