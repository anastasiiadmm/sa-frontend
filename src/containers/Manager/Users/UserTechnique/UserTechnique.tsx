import { Button, Table, Typography } from 'antd';
import { ColumnsType, TableProps } from 'antd/es/table';
import bem from 'easy-bem';
import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import arrowLeft from 'assets/images/icons/arrow-left.svg';
import deleteIcon from 'assets/images/icons/delete.svg';
import edit from 'assets/images/icons/edit.svg';
import tractorBlue from 'assets/images/icons/tractor-blue.svg';
import tractor from 'assets/images/icons/tractor-image.svg';
import AddUpdateTechnique from 'components/ModalComponent/ModalChildrenComponents/AddUpdateTechnique/AddUpdateTechnique';
import ModalComponent from 'components/ModalComponent/ModalComponent';
import 'containers/Manager/Users/UserTechnique/_userTechnique.scss';
import DeleteRejectTechniqueModal from 'components/ModalComponent/ModalChildrenComponents/DeleteTechniqueModal/DeleteTechniqueModal';

const { Title } = Typography;

interface DataType {
  key: React.Key;
  code: string;
  name: string;
  fields: string;
  area: string;
}

const UserTechnique: React.FC = () => {
  const b = bem('UserTechnique');
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
  const [isDeleteModalOpen, setIsDeleteEditModalOpen] = useState(false);

  const showModal = () => {
    setIsModalOpen(true);
  };

  const handleOk = () => {
    setIsModalOpen(false);
  };

  const handleCancel = () => {
    setIsModalOpen(false);
  };

  const showEditModal = () => {
    setIsEditModalOpen(true);
  };

  const handleEditOk = () => {
    setIsEditModalOpen(false);
  };

  const handleEditCancel = () => {
    setIsEditModalOpen(false);
  };

  const showDeleteModal = () => {
    setIsDeleteEditModalOpen(true);
  };

  const handleDeleteOk = () => {
    setIsDeleteEditModalOpen(false);
  };

  const handleDeleteCancel = () => {
    setIsDeleteEditModalOpen(false);
  };

  const deleteTechniqueHandler = () => {};

  const columns: ColumnsType<DataType> = [
    {
      title: 'Код техники',
      dataIndex: 'code',
      width: '20%',
      sorter: true,
      fixed: 'left',
      render: (text: string) => (
        <div style={{ display: 'flex', gap: 12 }}>
          <img src={tractor} alt='tractor' />
          <p className={b('name-column-style')}>{text}</p>
        </div>
      ),
    },
    {
      title: 'Название',
      dataIndex: 'name',
      filterSearch: true,
      width: '20%',
      sorter: true,
    },
    {
      title: 'Поля',
      dataIndex: 'fields',
      filterSearch: true,
      width: '20%',
      sorter: true,
    },
    {
      title: 'Общая Площадь',
      dataIndex: 'area',
      filterSearch: true,
      width: '20%',
      sorter: true,
    },
    {
      dataIndex: 'profile',
      filterSearch: true,
      width: '40%',
      render: () => (
        <div style={{ display: 'flex', gap: 10, alignItems: 'center' }}>
          <Link className={b('profile-link')} to='/user-profile'>
            Просмотр на карте
          </Link>
          <Button type='text' onClick={showEditModal}>
            <img src={edit} alt='edit' className='link-icons' />
          </Button>
          <Link to='/profile-technique'>
            <Button type='text'>
              <img src={tractorBlue} alt='tractorBlue' className={b('tractor-blue link-icons')} />
            </Button>
          </Link>
          <Button type='text' onClick={showDeleteModal}>
            <img src={deleteIcon} alt='deleteIcon' className='link-icons' />
          </Button>
        </div>
      ),
    },
  ];

  const data: DataType[] = [
    {
      key: '1',
      code: 'AVP123344',
      name: 'Беларус',
      fields: '3',
      area: '120 га',
    },
    {
      key: '2',
      code: 'AVP123344',
      name: 'Беларус',
      fields: '3',
      area: '120 га',
    },
    {
      key: '3',
      code: 'AVP123344',
      name: 'Беларус',
      fields: '3',
      area: '120 га',
    },
  ];

  const onChange: TableProps<DataType>['onChange'] = (pagination, filters, sorter, extra) => {};

  return (
    <>
      <div className={b()}>
        <div className={b('table')}>
          <div className={b('header')}>
            <div className={b('header-title')}>
              <Link to='/user-profile'>
                <img className={b('arrow-left')} src={arrowLeft} alt='arrow' />
              </Link>
              <Title level={3} className={b('title')}>
                Техника пользователя - <p className={b('subtitle')}> Иванов И.И</p>
              </Title>
            </div>

            <div>
              <Button type='primary' onClick={showModal}>
                Добавить технику
              </Button>
            </div>
          </div>

          <Table
            scroll={{
              x: 950,
            }}
            columns={columns}
            dataSource={data}
            onChange={onChange}
          />
        </div>
      </div>

      <ModalComponent
        title='Добавить технику'
        open={isModalOpen}
        handleOk={handleOk}
        handleCancel={handleCancel}
      >
        <AddUpdateTechnique />
      </ModalComponent>

      <ModalComponent
        title='Редактировать технику'
        open={isEditModalOpen}
        handleOk={handleEditOk}
        handleCancel={handleEditCancel}
      >
        <AddUpdateTechnique isEdit />
      </ModalComponent>

      <ModalComponent
        dividerShow={false}
        open={isDeleteModalOpen}
        handleOk={handleDeleteOk}
        handleCancel={handleDeleteCancel}
      >
        <DeleteRejectTechniqueModal
          title='Удалить?'
          subTitle='Вы уверены, что хотите удалить'
          techniqueName='Камаз 6595?'
          handleDeleteCancel={handleDeleteCancel}
          deleteRejectTechniqueHandler={deleteTechniqueHandler}
        />
      </ModalComponent>
    </>
  );
};

export default UserTechnique;
