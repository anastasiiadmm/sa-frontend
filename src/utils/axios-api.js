import axios from "axios";
import {apiURL} from "utils/config";

export const axiosApi = axios.create({
  baseURL: apiURL,
});