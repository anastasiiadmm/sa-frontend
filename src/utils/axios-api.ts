import axios from 'axios';

import { checkForTokens, logoutUser } from 'redux/auth/authSlice';
import { apiURL } from 'utils/config';
import { logoutLocalStorage } from 'utils/token';
import store from 'redux/store';

const axiosApi = axios.create({
  baseURL: apiURL,
});

axiosApi.interceptors.request.use(async (config) => {
  const key = store.getState()?.auth?.tokens?.access;

  if (key) {
    config.headers = {
      ...config.headers,
      Authorization: `Bearer ${key}`,
      'Content-Type': 'application/json',
    } as any;
  }
  return config;
});

axiosApi.interceptors.response.use(
  async (config) => {
    return config;
  },
  async (error) => {
    const originalRequest = error.config;

    const statusCode = error?.response?.status;
    const index = store;
    const { tokens } = index.getState().auth;
    if (
      tokens &&
      statusCode === 401 &&
      error.config &&
      !error.config._isReady &&
      error.response.data.messages
    ) {
      originalRequest._isReady = true;
      try {
        const resp = await axiosApi.post('/accounts/refresh/', { refresh: tokens.refresh });
        if (resp.status === 200) {
          const newTokens = resp.data;
          axiosApi.defaults.headers.Authorization = `Bearer ${newTokens.access}`;
          const usersLocal = {
            user: index.getState()?.auth?.user,
            token: {
              access: resp.data.access,
              refresh: tokens.refresh,
            },
          };
          index.dispatch(checkForTokens(usersLocal));
          const obj: any = usersLocal;
          delete obj.token.refresh;
          localStorage.setItem('users', JSON.stringify(obj));
          window.dispatchEvent(new Event('storage'));
          return axiosApi(originalRequest);
        }
      } catch (e) {
        logoutLocalStorage();
        window.location.reload();
      }

      store.dispatch(logoutUser());
    }

    return Promise.reject(error);
  },
);

export default axiosApi;
