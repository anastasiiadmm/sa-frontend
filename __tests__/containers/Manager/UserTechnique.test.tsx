import { BrowserRouter } from "react-router-dom";
import { cleanup } from "@testing-library/react";
import "../../../__mocks__/matchMedia.mock";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { mockedDispatch, mockedUseSelectors } from "../../../__mocks__/utils";

import UserTechnique from "../../../src/containers/Manager/Users/UserTechnique/UserTechnique";

afterEach(cleanup);

const data = {
    vehicleList: [
        {
            "id": 91,
            "code": "D8BN8TR61",
            "description": "fbd",
            "jobs_number": 1,
            "area": "927.006",
            "image": "/api/static/default_vehicle_image.jpg"
        }
    ]
};


describe("<UserTechnique />", () => {
    test("Users table component should be in the document", async () => {
        mockedUseSelectors.mockReturnValue(data);
        const dispatch = jest.fn();
        mockedDispatch.mockReturnValue(dispatch);

        const { asFragment } = render(
          <BrowserRouter>
              <UserTechnique />
          </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
