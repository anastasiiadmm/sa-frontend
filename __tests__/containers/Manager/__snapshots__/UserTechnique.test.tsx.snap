// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserTechnique /> Users table component should be in the document 1`] = `
<div
  className="UserTechnique"
  data-testid="user-technique-id"
>
  <div
    className="HeaderMobile"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="arrow"
        className="UserTechnique__arrow-left"
        src={
          {
            "process": [Function],
          }
        }
      />
    </button>
    <span>
      Техника пользователя
    </span>
  </div>
  <div
    className="UserTechnique__table"
  >
    <div
      className="UserTechnique__header"
    >
      <div
        className="UserTechnique__header-title"
      >
        <a
          href="/user-profile/undefined"
          onClick={[Function]}
        >
          <img
            alt="arrow"
            className="UserTechnique__arrow-left"
            src={
              {
                "process": [Function],
              }
            }
          />
        </a>
        <h3
          className="ant-typography UserTechnique__title css-dev-only-do-not-override-17a39f8"
          style={
            {
              "WebkitLineClamp": undefined,
            }
          }
        >
          Техника пользователя -
           
          <p
            className="UserTechnique__subtitle"
          >
             
            .
             
            undefined.
          </p>
        </h3>
      </div>
      <div
        className="UserTechnique__mobile_block"
      >
        <button
          className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default"
          disabled={false}
          onClick={[Function]}
          style={{}}
          type="button"
        >
          <div>
            <p>
              <img
                alt="tractorProfileIcon"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </p>
            <p>
              Добавить технику
            </p>
          </div>
        </button>
        <button
          className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default"
          disabled={false}
          onClick={[Function]}
          style={{}}
          type="button"
        >
          <div>
            <p>
              <img
                alt="tractorProfileIcon"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </p>
            <p>
              Посмотреть всю 
              <br />
               технику
            </p>
          </div>
        </button>
      </div>
      <div
        className="UserTechnique__btn_handler"
      >
        <button
          className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          style={{}}
          type="button"
        >
          <span>
            Добавить технику
          </span>
        </button>
        <a
          href="/technique-map/undefined"
          onClick={[Function]}
        >
          <button
            className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default UserTechnique__view-button"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <span>
              Вся техника на карте
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      className="UserTechnique__table_list"
    >
      <div
        className="ant-table-wrapper css-dev-only-do-not-override-17a39f8"
        style={{}}
      >
        <div
          className="ant-spin-nested-loading css-dev-only-do-not-override-17a39f8"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-scroll-horizontal"
            >
              <div
                className="ant-table-container"
              >
                <div
                  className="ant-table-content"
                  onScroll={[Function]}
                  style={
                    {
                      "overflowX": "auto",
                      "overflowY": "hidden",
                    }
                  }
                >
                  <table
                    style={
                      {
                        "minWidth": "100%",
                        "tableLayout": "auto",
                        "width": 950,
                      }
                    }
                  >
                    <colgroup>
                      <col
                        style={
                          {
                            "width": "20%",
                          }
                        }
                      />
                      <col
                        style={
                          {
                            "width": "20%",
                          }
                        }
                      />
                      <col
                        style={
                          {
                            "width": "20%",
                          }
                        }
                      />
                      <col
                        style={
                          {
                            "width": "45%",
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          scope="col"
                          style={{}}
                        >
                          Код техники
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          scope="col"
                          style={{}}
                        >
                          Название
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          scope="col"
                          style={{}}
                        >
                          Задача на обработку
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          scope="col"
                          style={{}}
                        >
                          Общая Площадь
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        className="ant-table-measure-row"
                        style={
                          {
                            "fontSize": 0,
                            "height": 0,
                          }
                        }
                      >
                        <td
                          style={
                            {
                              "border": 0,
                              "height": 0,
                              "padding": 0,
                            }
                          }
                        >
                          <div
                            style={
                              {
                                "height": 0,
                                "overflow": "hidden",
                              }
                            }
                          >
                             
                          </div>
                        </td>
                        <td
                          style={
                            {
                              "border": 0,
                              "height": 0,
                              "padding": 0,
                            }
                          }
                        >
                          <div
                            style={
                              {
                                "height": 0,
                                "overflow": "hidden",
                              }
                            }
                          >
                             
                          </div>
                        </td>
                        <td
                          style={
                            {
                              "border": 0,
                              "height": 0,
                              "padding": 0,
                            }
                          }
                        >
                          <div
                            style={
                              {
                                "height": 0,
                                "overflow": "hidden",
                              }
                            }
                          >
                             
                          </div>
                        </td>
                        <td
                          style={
                            {
                              "border": 0,
                              "height": 0,
                              "padding": 0,
                            }
                          }
                        >
                          <div
                            style={
                              {
                                "height": 0,
                                "overflow": "hidden",
                              }
                            }
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key={91}
                        onClick={[Function]}
                        style={{}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={{}}
                        >
                          <div
                            style={
                              {
                                "display": "flex",
                                "gap": 12,
                              }
                            }
                          >
                            <img
                              alt="tractor"
                              className="UserTechnique__image-styles"
                              src="https://localhost:8000/api/static/default_vehicle_image.jpg"
                            />
                            <p
                              className="UserTechnique__name-column-style"
                            >
                              D8BN8TR61
                            </p>
                          </div>
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={{}}
                        >
                          fbd
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={{}}
                        >
                          <p>
                            1
                          </p>
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          rowSpan={null}
                          style={{}}
                        >
                          <div
                            style={
                              {
                                "alignItems": "center",
                                "display": "flex",
                                "justifyContent": "right",
                              }
                            }
                          >
                            <p
                              className="UserTechnique__text_processed_area"
                            >
                              927.006
                            </p>
                            <button
                              className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
                              disabled={false}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onPointerEnter={[Function]}
                              onPointerLeave={[Function]}
                              style={{}}
                              type="button"
                            >
                              <img
                                alt="edit"
                                className="link-icons"
                                src={
                                  {
                                    "process": [Function],
                                  }
                                }
                              />
                            </button>
                            <a
                              href="/profile-technique/91"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onPointerEnter={[Function]}
                              onPointerLeave={[Function]}
                            >
                              <button
                                className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
                                disabled={false}
                                onClick={[Function]}
                                style={{}}
                                type="button"
                              >
                                <img
                                  alt="tractorBlue"
                                  className="UserTechnique__tractor-blue link-icons"
                                  src={
                                    {
                                      "process": [Function],
                                    }
                                  }
                                />
                              </button>
                            </a>
                            <a
                              href="/open-map/91/local-tractor/D8BN8TR61"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onPointerEnter={[Function]}
                              onPointerLeave={[Function]}
                            >
                              <button
                                className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
                                disabled={false}
                                onClick={[Function]}
                                style={{}}
                                type="button"
                              >
                                <img
                                  alt="tractorBlue"
                                  className="UserTechnique__tractor-blue link-icons"
                                  src={
                                    {
                                      "process": [Function],
                                    }
                                  }
                                />
                              </button>
                            </a>
                            <button
                              className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table UserTechnique__deleteBg"
                              disabled={false}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onPointerEnter={[Function]}
                              onPointerLeave={[Function]}
                              style={{}}
                              type="button"
                            >
                              <img
                                alt="deleteIcon"
                                className="link-icons"
                                src={
                                  {
                                    "process": [Function],
                                  }
                                }
                              />
                            </button>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pagination"
      >
        <div
          className="pagination__pagination_show"
        >
          <div
            className="glav_pagination"
          >
            <div
              className="next_table"
            >
              <button
                className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-prev"
                disabled={true}
                onClick={[Function]}
                style={{}}
                type="button"
              >
                <span
                  className="ant-btn-icon"
                  style={{}}
                >
                  <img
                    alt="backIcons"
                    src={
                      {
                        "process": [Function],
                      }
                    }
                  />
                </span>
              </button>
              <button
                className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-next"
                disabled={true}
                onClick={[Function]}
                role="button"
                style={{}}
                type="button"
              >
                <span
                  className="ant-btn-icon"
                  style={{}}
                >
                  <img
                    alt="nextIcons"
                    src={
                      {
                        "process": [Function],
                      }
                    }
                  />
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="UserTechnique__list_table_mobile"
  >
    <div
      className="UserTechnique__table_mobile"
    >
      <div
        className="block_avatar"
      >
        <div>
          <img
            src="https://localhost:8000/api/static/default_vehicle_image.jpg"
          />
        </div>
        <div
          className="block_title"
        >
          <p>
            <b>
              Название техники
            </b>
          </p>
          <p
            className="todo"
          >
            fbd
          </p>
          <p
            className="todo"
          >
            Задачи 
            <b>
              1
            </b>
          </p>
        </div>
      </div>
      <div
        style={
          {
            "display": "flex",
            "marginTop": 10,
          }
        }
      >
        <button
          className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPointerEnter={[Function]}
          onPointerLeave={[Function]}
          style={{}}
          type="button"
        >
          <img
            alt="edit"
            className="link-icons"
            src={
              {
                "process": [Function],
              }
            }
          />
        </button>
        <a
          href="/profile-technique/91"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPointerEnter={[Function]}
          onPointerLeave={[Function]}
        >
          <button
            className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <img
              alt="tractorBlue"
              className="UserTechnique__tractor-blue link-icons"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </button>
        </a>
        <a
          href="/open-map/91/local-tractor/D8BN8TR61"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPointerEnter={[Function]}
          onPointerLeave={[Function]}
        >
          <button
            className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <img
              alt="tractorBlue"
              className="UserTechnique__tractor-blue link-icons"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </button>
        </a>
        <button
          className="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text UserTechnique__btn_table UserTechnique__deleteBg"
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPointerEnter={[Function]}
          onPointerLeave={[Function]}
          style={{}}
          type="button"
        >
          <img
            alt="deleteIcon"
            className="link-icons"
            src={
              {
                "process": [Function],
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
